build/xtask/src/clippy.rs

```diff
@@ -47,7 +47,17 @@ pub fn run(
         }
 
         let build_config = if name == "kernel" {
-            let (allocs, _) = crate::dist::allocate_all(&toml)?;
+            // Build dummy allocations for each task
+            let task_sizes = toml
+                .tasks
+                .keys()
+                .map(|name| name.as_str())
+                .zip(std::iter::repeat_with(|| {
```

Comment:
I'm an iterator nerd, but, I had to stare at this for a bit. What about:

```rust
// above
let fake_sizes: HashMap<_, _> = [("flash", 64), ("ram", 64)].into_iter().collect();

// below that
.map(|name| (name.as_str(), fake_sizes.clone())
```

instead of map + zip + repeat_with?

Alternatively, based on your use of the `allocate_all` function, I'd consider changing its `task_sizes` argument type to

`impl FnMut(&str) -> u64`

and then for _this_ use you can pass in `|_| 64`, whereas for the original use, it'd be `|name| task_sizes[name]`

Reply:
Yeah, I'll do the simpler option (of cloning a separate `fake_sizes` variable)